### PR TITLE
Fix: Navigation between list and isolated mode issues

### LINKED ISSUE #663: Navigation is broken
Reproducible on the public example styleguide: https://react-styleguidist.js.org/examples/basic/
~~Not reproducible on `dialog`'s styleguide: https://dialogs.github.io/dialog-web-components/~~ Can also repro there

Issue is that navigation is super broken. See the GIF:

out

- Navigating to a preview and then back results in other previews from the same section disappearing
- Trying to navigate from a preview back to the main page by clicking the little icon doesn't work
- Navigating directly to a preview by changing the URL results in a nasty error
- Back button on the browser seems to do nothing

Why does this not reproduce on `dialog`'s styleguide? How is it different from your basic example?

### PR DESCRIPTION
**Pull request by sapegin**
1. Navigate to isolated example and back: you’ll see only this one
example, not all of them.
2. Navigate to isolated example of a component with only one example:
you’ll see isolated mode button below this example instead of a show
all components button.

Browser back / forward buttons also seems to work.

Closes #663, #467

### CODE REVIEW

**Comment by unindented** [comment]
Thank you!

File: src/rsg-components/Section/Section.spec.js [hunk 1]
```diff
@@ -6,6 +6,12 @@ import Sections from '../Sections';
 import Section from './Section';
 import { SectionRenderer } from './SectionRenderer';
 
+const options = {
+	context: {
+		displayMode: 'all',
```
Comment (okonet) [inline src/rsg-components/Section/Section.spec.js]: I'd create a enum (`Object.freeze`) with all available modes and import it any time you reference it.

File: src/rsg-components/Section/Section.js [hunk 2]
```diff
@@ -31,5 +31,5 @@ Section.propTypes = {
 };
 
 Section.contextTypes = {
-	isolatedSection: PropTypes.bool,
+	displayMode: PropTypes.string,
```
Comment (okonet) [inline src/rsg-components/Section/Section.js]: Instead of saying `string` it's better to use `oneOf` propType with exhausting set of values.

**Comment by sapegin** [comment]
Yeah, good points. I decided to work on that more and was able isolate, fix and test mutation instead of using a hack with `cloneDeep`.

By the way how would you ensure that you’re not mutating source objects in tests?

**Comment by okonet** [comment]
Not sure about your questions. My points are more about the maintainability. What do you mean by mutating in test? Example?

**Comment by sapegin** [comment]
I mean tests to prevent mutation in code. That was exactly the issue here.

**Comment by okonet** [comment]
Ah no, I don't know how to tests side-effects.

File: src/rsg-components/Section/Section.spec.js (hunk 1 above)
Comment (sapegin) [reply src/rsg-components/Section/Section.spec.js]: Do you have an example somewhere?

**Comment by sapegin** [comment]
I’ve added deepfreeze to tests, so it should catch mutations. And I’ve removed cloneDeep hack. 

I still need to replace strings with an enum.

**Comment by sapegin** [comment]
Merged manually.

**Pull request closed without merging**